diff -ruwN source/apps/lib/apps.c source-new/apps/lib/apps.c
--- source/apps/lib/apps.c	2025-09-16 19:05:33.000000000 +0700
+++ source-new/apps/lib/apps.c	2026-05-05 05:08:40.007422945 +0700
@@ -2839,7 +2839,7 @@
     return ret;
 }
 
-#elif defined(_SC_CLK_TCK)      /* by means of unistd.h */
+#elif defined(_SC_CLK_TCK) && !defined(__RELIBC__)      /* by means of unistd.h */
 # include <sys/times.h>
 
 double app_tminterval(int stop, int usertime)
diff -ruwN source/Configurations/10-main.conf source-new/Configurations/10-main.conf
--- source/Configurations/10-main.conf	2025-09-16 19:05:33.000000000 +0700
+++ source-new/Configurations/10-main.conf	2026-05-05 05:08:44.812616321 +0700
@@ -208,6 +208,40 @@
         shared_extension => ".so",
     },
 
+### Redox configurations
+    "redox-common" => {
+        inherit_from     => [ "BASE_unix" ],
+        cc               => "gcc",
+        thread_scheme    => "pthreads",
+        dso_scheme       => "dlfcn",
+        shared_target    => "linux-shared",
+        shared_cflag     => "-fPIC",
+        shared_ldflag    => sub { $disabled{pinshared} ? () : "-Wl,-znodelete" },
+        shared_extension => ".so.\$(SHLIB_VERSION_NUMBER)",
+    },
+    "redox-generic64" => {
+        inherit_from     => [ "redox-common" ],
+        bn_ops           => "SIXTY_FOUR_BIT_LONG RC4_CHAR",
+    },
+    "redox-aarch64" => {
+        inherit_from     => [ "redox-generic64" ],
+        perlasm_scheme   => "linux64",
+    },
+    "redox-i586" => {
+        inherit_from     => [ "redox-common" ],
+        bn_ops           => "BN_LLONG",
+        perlasm_scheme   => "elf",
+    },
+    "redox-riscv64gc" => {
+        inherit_from     => [ "redox-generic64" ],
+        perlasm_scheme   => "linux64",
+    },
+    "redox-x86_64" => {
+        inherit_from     => [ "redox-generic64" ],
+        perlasm_scheme   => "elf",
+    },
+
+
 #### Solaris configurations
     "solaris-common" => {
         inherit_from     => [ "BASE_unix" ],
@@ -693,7 +727,6 @@
         shared_target    => "linux-shared",
         shared_cflag     => "-fPIC",
         shared_ldflag    => sub { $disabled{pinshared} ? () : "-Wl,-znodelete" },
-        enable           => [ "afalgeng" ],
     },
     "linux-latomic" => {
         inherit_from     => [ "linux-generic32" ],
@@ -890,7 +923,6 @@
         bn_ops           => "SIXTY_FOUR_BIT_LONG",
         asm_arch         => 'x86_64',
         perlasm_scheme   => "elf",
-        multilib         => "64",
     },
     "linux-x86_64-clang" => {
         inherit_from     => [ "linux-x86_64" ],
diff -ruwN source/include/crypto/rand.h source-new/include/crypto/rand.h
--- source/include/crypto/rand.h	2025-09-16 19:05:33.000000000 +0700
+++ source-new/include/crypto/rand.h	2026-05-05 05:08:40.007980910 +0700
@@ -41,7 +41,7 @@
  * default, we will try to read at least one of these files
  */
 # define DEVRANDOM "/dev/urandom", "/dev/random", "/dev/hwrng", "/dev/srandom"
-# if defined(__linux) && !defined(__ANDROID__)
+# if defined(__linux) && !defined(__RELIBC__) && !defined(__ANDROID__) /* needs sys/shm.h */
 #  ifndef DEVRANDOM_WAIT
 #   define DEVRANDOM_WAIT   "/dev/random"
 #  endif
