diff -ruwN source/include/X11/Xfuncs.h source-new/include/X11/Xfuncs.h
--- source/include/X11/Xfuncs.h	2024-03-26 14:16:32.000000000 +0700
+++ source-new/include/X11/Xfuncs.h	2026-05-02 08:17:20.538178973 +0700
@@ -44,7 +44,7 @@
 #    define bcmp(b1,b2,len) memcmp(b1, b2, len)
 #   else
 #    include <string.h>
-#    if defined(__SCO__) || defined(__sun) || defined(__UNIXWARE__) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__)
+#    if defined(__SCO__) || defined(__sun) || defined(__UNIXWARE__) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__) || defined(__redox__)
 #     include <strings.h>
 #    endif
 #    define _XFUNCS_H_INCLUDED_STRING_H
diff -ruwN source/include/X11/Xos.h source-new/include/X11/Xos.h
--- source/include/X11/Xos.h	2024-03-26 14:16:32.000000000 +0700
+++ source-new/include/X11/Xos.h	2026-05-02 08:18:39.611339183 +0700
@@ -60,7 +60,7 @@
  */
 
 # include <string.h>
-# if defined(__SCO__) || defined(__UNIXWARE__) || defined(__sun) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__) || defined(__FreeBSD__)
+# if defined(__SCO__) || defined(__UNIXWARE__) || defined(__sun) || defined(__CYGWIN__) || defined(_AIX) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__redox__)
 #  include <strings.h>
 # else
 #  ifndef index
diff -ruwN source/include/X11/Xos_r.h source-new/include/X11/Xos_r.h
--- source/include/X11/Xos_r.h	2024-03-26 14:16:32.000000000 +0700
+++ source-new/include/X11/Xos_r.h	2026-05-02 08:17:20.538337051 +0700
@@ -318,7 +318,7 @@
   (_Xos_processUnlock), \
   (p).pwp )
 
-#elif !defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(__APPLE__)
+#elif !defined(_POSIX_THREAD_SAFE_FUNCTIONS) && !defined(__APPLE__) && !defined(__redox__)
 # define X_NEEDS_PWPARAMS
 typedef struct {
   struct passwd pws;
